DNN Platform/Library/Security/PortalSecurity.cs

```diff
@@ -37,6 +37,8 @@ public class PortalSecurity
 
         private static readonly Regex StripTagsRegex = new Regex("<[^<>]*>", RegexOptions.IgnoreCase | RegexOptions.Singleline | RegexOptions.Compiled);
         private static readonly Regex BadStatementRegex = new Regex(BadStatementExpression, RegexOptions.IgnoreCase | RegexOptions.Compiled);
+        private static readonly Regex ControlCharactersRegex = new Regex(@"\p{C}+", RegexOptions.Compiled);
+        private static readonly Regex ControlCharacterToWhitespaceRegex = new Regex(@"\r\n|\r|\n|\t", RegexOptions.Compiled);

```

Comment:
If we're matching `\r` and `\n`, does it make a difference to search for them together also?

Reply:
Oh, you're trying to avoid introducing multiple spaces, got it.

Reply:
Windows uses `\r\n` for new lines and `\r\n` is seen as one new line, instead of two. If we were to make this case separate, they would show two spaces. If this isn't a problem, I can remove `\r\n`.